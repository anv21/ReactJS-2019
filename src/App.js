import React from 'react';
import './assets/style/index.css';
import {connect} from 'react-redux';
import PropTypes from 'prop-types';

import Header from './components/Header/index.jsx';
import Items from './components/Items/index.jsx';
import Footer from './components/Footer/index.jsx';
import ErrorBoundary from './components/ErrorBoundary.jsx';

import {
    getItems,
    getItem,
    getItemsByGenre,
    setSearchValue,
    setSortBy,
    setSearchBy,
} from './actions/index.js';

export class App extends React.Component {
    componentWillMount() {
        const {getItems} = this.props;
        getItems();
    }

    render() {
        return (
            <ErrorBoundary>
                <Header {...this.props} />
                <Items {...this.props} />
                <Footer />
            </ErrorBoundary>
        )
    }
}

Items.propTypes = {
    getItems: PropTypes.func,
};

Items.defaultProps = {
    getItems: null,
};

const mapStateToProps = (state) => {
    const {
        items,
        item,
        appState,
        total,
        value,
        sortBy,
        searchBy
    } = state.appReducer;

     return {
        items,
        item,
        appState,
        total,
        value,
        sortBy,
        searchBy
    };
};

 export default connect(mapStateToProps, {
    getItems,
    getItem,
    getItemsByGenre,
    setSearchValue,
    setSortBy,
    setSearchBy,
})(App);