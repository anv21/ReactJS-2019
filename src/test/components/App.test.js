import React from 'react';
import { shallow } from 'enzyme';

import {App} from '../../App.js';

describe('App', () => {
    const mockGetItems = jest.fn();
    const app = shallow(<App getItems={mockGetItems} />);

    it('should be like snapshot', () => {
        expect(app).toMatchSnapshot();
    });
});