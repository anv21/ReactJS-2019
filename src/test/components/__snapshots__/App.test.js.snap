// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be like snapshot 1`] = `ShallowWrapper {}`;
